repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeFilter.scala

```diff
@@ -19,7 +19,193 @@
+        val traceGUID = Option(wrappedRequest.getHeader(TRACE_GUID)) match {
+          case Some(string) if StringUtils.isNotBlank(string) => string
+          case _ => UUID.randomUUID.toString
+        }
+
+        MDC.put(TracingKey.TRACING_KEY, traceGUID)
+
+        try {
+          TryWith(filterContextList)(filterContextListToo => {
+            // Ensure the request URI is a valid URI
+            // This object is only being created to ensure its validity.
+            // So it is safe to suppress warning squid:S1848
+            new URI(wrappedRequest.getRequestURI)
+
+            if (tracingHeaderConfig.exists(_.isEnabled)) {
+              if (StringUtils.isBlank(wrappedRequest.getHeader(TRACE_GUID))) {
+                wrappedRequest.addHeader(TRACE_GUID, TracingHeaderHelper.createTracingHeader(traceGUID, wrappedRequest.getHeader(VIA)))
+              }
+              if (tracingHeaderConfig.exists(_.isSecondaryPlainText)) {
+                TraceIdLogger.trace("Adding plain text trans id to request: {}", traceGUID)
+                wrappedRequest.replaceHeader(REQUEST_ID, traceGUID)
+              }
+              val tracingHeader = wrappedRequest.getHeader(TRACE_GUID)
+              TraceIdLogger.info("Tracing header: {}", TracingHeaderHelper.decode(tracingHeader))
+              wrappedResponse.addHeader(TRACE_GUID, tracingHeader)
+            }
+            new ReposeFilterChain(
+              filterContextListToo.filterContexts,
+              chain,
+              None,
+              optMetricRegistry,
+              tracer
+            ).doFilter(wrappedRequest, wrappedResponse)
+          }) match {
+            case Success(_) =>
+              ThisLogger.trace("{} -- Successfully processed request", nodeId)
+            case Failure(use: URISyntaxException) =>
+              ThisLogger.debug(s"$nodeId -- Invalid URI requested: ${wrappedRequest.getRequestURI}", use)
+              wrappedResponse.sendError(SC_BAD_REQUEST, "Error processing request")
+            case Failure(e: Exception) =>
+              ThisLogger.error(s"$nodeId -- Issue encountered while processing filter chain.", e)
+              wrappedResponse.sendError(SC_BAD_GATEWAY, "Error processing request")
+            case Failure(t: Throwable) =>
+              ThisLogger.error(s"$nodeId -- Error encountered while processing filter chain.", t)
+              wrappedResponse.sendError(SC_BAD_GATEWAY, "Error processing request")
+              throw t
+          }
+        } finally {
+          closeSpan(wrappedResponse, scope)
+          if (!wrappedResponse.isCommitted) {
+            responseHeaderService.setVia(wrappedRequest, wrappedResponse)
+          }
+          wrappedResponse.commitToResponse()
+          optMetricRegistry.foreach { (mr: MetricRegistry) =>
+            markResponseCodeHelper(
+              mr,
+              wrappedResponse.asInstanceOf[HttpServletResponse].getStatus,
```

Comment:
if its wrapped there should be zero need to cast it.

Reply:
True. Didn't catch that one on the initial conversion from Java.